File: src/main/java/com/collicode/algodatastructure/leetcode/twopointers/ContainerWithMostWater.java
package com.collicode.algodatastructure.leetcode.twopointers;

public class ContainerWithMostWater {


    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        int maxArea = 0;
        while (left < right) {
            int width = right - left;
            int height1 = Math.min(height[left], height[right]);
            maxArea = Math.max(maxArea, width * height1);

            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }

        }
        return maxArea;

    }
}

File: src/main/java/com/collicode/algodatastructure/prep/week1/kotlin/ContainerWithMostWater.kt
package com.collicode.algodatastructure.prep.week1.kotlin


class ContainerWithMostWater {

    fun maxArea(height: IntArray): Int {
        var left = 0
        var right = height.size - 1
        var maxArea = 0

        while (left < right) {
            val width = right - left
            val minHeight = minOf(height[left], height[right])

            maxArea = maxOf(maxArea, width * minHeight)

            if (height[left] < height[right]) {
                left++
            } else {
                right--
            }
        }

        return maxArea
    }
}